lib/ui/painting.dart

```diff
@@ -1182,6 +1182,20 @@ class Paint {
     }
   }
 
+  @override
+  int get hashCode => hashValues(hashList(_data.buffer.asUint8List()), hashList(_objects));
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! Paint)
+      return false;
+    final Paint typedOther = other;
+    return equalList(_data.buffer.asUint8List(), typedOther._data.buffer.asUint8List()) &&
+           equalList(_objects, typedOther._objects);
+  }
+
```

Comment:
I think we probably don't want to do this using operator ==, because they are mutable. It's a bit weird that two objects might sometimes be == and sometimes not.

We could have a method that reports if two paints are identical, though, if that's useful for this patch.

Reply:
This is especially a problem since this patch adds a Paint to TextStyle, which is immutable — it's extremely likely that parts of the framework might compare two TextStyles, decide they're identical and so drop one, and then later they will stop being identical because they're not actually the same Paint even though they briefly appeared the same.

Reply:
Removed the overloads of operator== and hashCode in Paint